app/models/patient_session.rb

```diff
@@ -141,9 +141,18 @@ def confirm_transfer!
     return unless pending_transfer?
 
     PatientSession.transaction do
-      PatientSession.create!(patient:, session: proposed_session)
-
-      safe_to_destroy? ? destroy! : update!(proposed_session: nil)
+      new_patient_session =
+        PatientSession.create!(patient:, session: proposed_session)
+
+      if safe_to_destroy?
+        school = new_patient_session.location
+        school = nil if school.generic_clinic?
+
+        patient.update!(school:)
```

Comment:
I think we should always update the school even if we can't delete the old session, in the case of a move, we'd still want the new school to be reflected on the patient.

Reply:
Yeah, makes sense